packages/shared/ReactTypeOfWork.js

```diff
@@ -28,18 +28,18 @@ export type TypeOfWork =
 
 export const IndeterminateComponent = 0; // Before we know whether it is functional or class
 export const FunctionalComponent = 1;
-export const ClassComponent = 2;
-export const HostRoot = 3; // Root of a host tree. Could be nested inside another node.
-export const HostPortal = 4; // A subtree. Could be an entry point to a different renderer.
-export const HostComponent = 5;
-export const HostText = 6;
-export const CallComponent_UNUSED = 7;
-export const CallHandlerPhase_UNUSED = 8;
-export const ReturnComponent_UNUSED = 9;
-export const Fragment = 10;
-export const Mode = 11;
-export const ContextConsumer = 12;
-export const ContextProvider = 13;
-export const ForwardRef = 14;
+export const FunctionalComponentLazy = 2;
+export const ClassComponent = 3;
+export const ClassComponentLazy = 4;
+export const HostRoot = 5; // Root of a host tree. Could be nested inside another node.
+export const HostPortal = 6; // A subtree. Could be an entry point to a different renderer.
+export const HostComponent = 7;
+export const HostText = 8;
+export const Fragment = 9;
+export const Mode = 10;
+export const ContextConsumer = 11;
+export const ContextProvider = 12;
+export const ForwardRef = 13;
+export const ForwardRefLazy = 14;
```

Comment:
DevTools won't be happy about this change

Reply:
Is there perhaps a way to expose this file to devtools, that way it can find & read it, as to not break?

Reply:
I think that hardcoding this in devtools based on version is probably the way to go. It doesn't require us to include extra strings and object initialization in the production builds.

I suspect that even the fiber fields might be mangled in the future so it'll need the same parity.

If we move to another language in the future, that language likely don't let us define our own indices. It'll be a header file or something that determines this based on compiler settings resulting in a reified contract. To make that work you typically have to compile both sides from the same source with the same settings. E.g. the devtools would have one header file per version.

That's the equivalent of the devtools just keeping a copy of this file per version. Maybe we should do that?

Reply:
Since 16 we [inject the version into DevTools](https://github.com/facebook/react/blob/5816829170662eed09e9681ce387d5672fc3da31/packages/react-dom/src/client/ReactDOM.js#L772) so yeah, we can do this.